From 7b62e5884353b247f542844d1e4687d0e9211999 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 20 Jul 2017 04:27:32 -0700
Subject: [PATCH 1/2] Check and use strlcpy from libc before defining own

This is required especially on musl where
function prototype conflicts and causes build
failures.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
Upstream-Status: Pending

 configure.ac | 2 +-
 lib/util.c   | 2 ++
 lib/util.h   | 1 +
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 13064f6..596c43f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -57,7 +57,7 @@ OFP_CHECK_HWTABLES
 OFP_CHECK_HWLIBS
 AC_SYS_LARGEFILE
 
-AC_CHECK_FUNCS([strsignal])
+AC_CHECK_FUNCS([strlcpy strsignal])
 
 AC_ARG_VAR(KARCH, [Kernel Architecture String])
 AC_SUBST(KARCH)
diff --git a/lib/util.c b/lib/util.c
index 21cc28d..1f341b1 100644
--- a/lib/util.c
+++ b/lib/util.c
@@ -138,6 +138,7 @@ xasprintf(const char *format, ...)
     return s;
 }
 
+#ifndef HAVE_STRLCPY
 void
 strlcpy(char *dst, const char *src, size_t size)
 {
@@ -148,6 +149,7 @@ strlcpy(char *dst, const char *src, size_t size)
         dst[n_copy] = '\0';
     }
 }
+#endif
 
 void
 ofp_fatal(int err_no, const char *format, ...)
diff --git a/lib/util.h b/lib/util.h
index fde681f..9e45ea9 100644
--- a/lib/util.h
+++ b/lib/util.h
@@ -41,6 +41,7 @@
 #include <stdio.h>
 #include <string.h>
 #include "compiler.h"
+#include "config.h"
 
 #ifndef va_copy
 #ifdef __va_copy
-- 
2.13.3

